### PR TITLE
Introduced quarkus-bootstrap-bom

### LINKED ISSUE #12628: copy dependency management for bootstrap modules to quarkus-bom
The quarkus-universe-bom can contain dependency management entries coming from extensions which override those used by the core quarkus bootstrap modules, as the core quarkus-bom may not contain dependency management entries for them while the platform quarkus-universe-bom could due to an extension doing so.

While looking into and discussing [1] an extension build failure I was seeing while trying the build out using Quarkus 1.9.0.CR1 core, due to an enforcer dependency convergence check failing for plexus-classworlds (which turned out to be a non-issue in the end, resolved by updating the Maven install on the little-used system from 3.6.0 to 3.6.3), I had also noticed that while the bootstrap modules have dependencyManagement meaning plexus-classworlds 2.6.0 should be used, the core quarkus-bom does not mention it, whilst the platform quarkus-universe-bom does but with the older 2.5.2 version.

During the discussion @aloubyansky suggested that the bootstrap dependency management should be copied to quarkus-bom and asked I raise an issue for it.

[1] https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/plexus-classworlds.20dep.20convergence.20with.201.2E9.2E0.2ECR1

### PR DESCRIPTION
**Pull request by aloubyansky**
Fixes #12628

This PR also makes sure the proper maven-settings artifact is used to parse the Maven settings file by the bootstrap module during the tests.

### CODE REVIEW

File: test-framework/common/log [hunk 1]
```diff
@@ -0,0 +1,112 @@
+[INFO] Scanning for projects...
```
Comment (famod) [inline test-framework/common/log:1]: Committed accidentially?

**Comment by famod** [comment]
I wonder if some deps should now be removed from the app BOM that are already in the bootstrap BOM, e.g. maven-resolver?

**Comment by aloubyansky** [comment]
> I wonder if some deps should now be removed from the app BOM that are already in the bootstrap BOM, e.g. maven-resolver?

Yes, I'll clean it up. Thanks.

**Review by famod — APPROVED** [review]
LGTM!
TBH, first I didn't really like the addition of yet another BOM, but after having digged myself for while, I now do also think that this is the only way to really enforce everything we need.

I'd just like to point out #9595, which introduced sisu to the app BOM. Maybe this should be revisited/moved as well.

**Comment by aloubyansky** [comment]
Thanks for reviewing @famod I moved the sisu inject to the bootstrap bom.

**Review by gsmet — APPROVED** [review]

**Comment by famod** [comment]
@aloubyansky this change causes a build warning (spotted locally);
```
[WARNING]
[WARNING] Some problems were encountered while building the effective model for io.quarkus:quarkus-bootstrap-bom:pom:999-SNAPSHOT
[WARNING] 'dependencyManagement.dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: io.quarkus:quarkus-bootstrap-core:jar -> duplicate declaration of version ${project.version} @ line 80, column 25
[WARNING] 'dependencyManagement.dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: io.quarkus:quarkus-bootstrap-app-model:jar -> duplicate declaration of version ${project.version} @ line 92, column 25
[WARNING] 'dependencyManagement.dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: io.quarkus:quarkus-bootstrap-maven-resolver:jar -> duplicate declaration of version ${project.version} @ line 97, column 25
[WARNING] 'dependencyManagement.dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: io.quarkus:quarkus-bootstrap-gradle-resolver:jar -> duplicate declaration of version ${project.version} @ line 107, column 25
[WARNING]
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING]
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING]
```

**Comment by aloubyansky** [comment]
Sorry about that https://github.com/quarkusio/quarkus/pull/18196

**Pull request merged**